streams/src/test/java/org/apache/kafka/streams/processor/internals/DefaultStateUpdaterTest.java

```diff
@@ -1012,6 +1170,22 @@ private void verifyExceptionsAndFailedTasks(final ExceptionAndTasks... exception
         );
     }
 
+    private void verifyFailedTasks(final Task... tasks) throws Exception {
+        final List<Task> expectedFailedTasks = Arrays.asList(tasks);
+        final Set<Task> failedTasks = new HashSet<>();
+        waitForCondition(
+                () -> {
+                    for (final ExceptionAndTasks exceptionsAndTasks : stateUpdater.getExceptionsAndFailedTasks()) {
+                        failedTasks.addAll(exceptionsAndTasks.getTasks());
+                    }
+                    return failedTasks.containsAll(expectedFailedTasks)
+                            && failedTasks.size() == expectedFailedTasks.size();
+                },
+                VERIFICATION_TIMEOUT,
+                "Did not get all exceptions and failed tasks within the given timeout!"
+        );
+    }
+
```

Comment:
Why do you introduce this verification method? For testing exceptions there is already `verifyExceptionsAndFailedTasks()`. You should anyways also verify the thrown exception and not only the failed tasks because otherwise any exception will satisfy this verification not only the expected one.

Reply:
`verifyExceptionsAndFailedTasks` verifies that the exception object is exactly the same, but here since we do not control the creation of the illegal state, we cannot do the exact match verification.

I augmented this function with the exception class to be differentiated with the other `verifyExceptionsAndFailedTasks.